hydromt/components/spatial.py

```diff
@@ -246,20 +210,117 @@ def write(
+        value0 = kwargs.pop(kind)
+
+        if kind in PLUGINS.model_plugins:
+            model_class = PLUGINS.model_plugins[kind]
+            other_model = model_class(root=value0, mode="r", logger=self._logger)
+            geom = other_model.region.region
+        # TODO: Move to MeshComponent
+        elif kind == "mesh":
+            if _compat.HAS_XUGRID:
+                if isinstance(value0, (str, Path)) and isfile(value0):
+                    kwarg = dict(mesh=xu.open_dataset(value0))
+                elif isinstance(value0, (xu.UgridDataset, xu.UgridDataArray)):
+                    kwarg = dict(mesh=value0)
+                elif isinstance(value0, (xu.Ugrid1d, xu.Ugrid2d)):
+                    kwarg = dict(
+                        mesh=xu.UgridDataset(
+                            value0.to_dataset(optional_attributes=True)
+                        )
+                    )
+                else:
+                    raise ValueError(
+                        f"Unrecognized type {type(value0)}."
+                        "Should be a path, data catalog key or xugrid object."
+                    )
+                kwargs.update(kwarg)
+            else:
+                raise ImportError("xugrid is required to read mesh files.")
+        elif kind not in options:
+            k_lst = '", "'.join(list(options.keys()))
+            raise ValueError(
+                f'Region key "{kind}" not understood, select from "{k_lst}"'
+            )
+
+        kwarg = _parse_region_value(value0, data_catalog=self._data_catalog)
+        if len(kwarg) == 0 or next(iter(kwarg)) not in options[kind]:
+            v_lst = '", "'.join(list(options[kind]))
+            raise ValueError(
+                f'Region value "{value0}" for kind={kind} not understood, '
+                f'provide one of "{v_lst}"'
+            )
+        kwargs.update(kwarg)
+
+        if kind in ["basin", "subbasin", "interbasin"]:
+            # retrieve global hydrography data (lazy!)
+            ds_org = self._data_catalog.get_rasterdataset(hydrography_fn)
+            if "bounds" not in region:
+                region.update(basin_index=self._data_catalog.get_source(basin_index_fn))
+            # get basin geometry
+            geom, _ = get_basin_geometry(
+                ds=ds_org,
+                kind=kind,
+                logger=self._logger,
+                **region,
+            )
+        elif "bbox" in region:
+            bbox = region["bbox"]
+            geom = gpd.GeoDataFrame(geometry=[box(*bbox)], crs=4326)
+        elif "geom" in region:
+            geom = region["geom"]
+            if geom.crs is None:
+                raise ValueError('Model region "geom" has no CRS')
+
+        if crs is not None and geom is not None:
+            crs = gis_utils.parse_crs(crs, bbox=geom.total_bounds)
+            geom = geom.to_crs(crs)
+
+        # TODO: This won't be called when subclassed. Should this move to _create_region?
```

Comment:
This TODO should be picked up before merging, although it's only logging.

Reply:
If the implementor inherits `parse_region`, and it does a early return, because e.g. `if kind == "grid"`, then we don't come through any of this code. And the logging seems to be a part that we would want to have, even if the result was a grid.

Reply:
Maybe better in create region indeed

Reply:
outdated comment